tests/integration/components/denali-tag-test.js

```diff
@@ -81,12 +87,34 @@ module('Integration | Component | denali-tag', function (hooks) {
       .hasClass('has-icon-front', 'DenaliTag has the `has-icon-front` class when iconFront is specified');
     assert.dom('span.tag .d-icon').hasClass('d-check', 'DenaliTag has the specified icon in the front');
 
+    const iconFrontClass = 'iconFrontClass';
+    assert
+      .dom('span.tag .d-icon')
+      .doesNotHaveClass(iconFrontClass, 'DenaliTag does not have `iconFrontClass` class by default');
+    this.set('iconFrontClass', iconFrontClass);
+    assert.dom('span.tag .d-check').hasClass(iconFrontClass, 'DenaliTag has the specified `iconFrontClass`');
+
+    this.set('onIconFrontClick', () => assert.ok(true, '`onIconFrontClick` fires on click'));
+    await click('.d-check');
+
     // unset the front icon to check the back icon
     this.set('iconFront', undefined);
+    this.set('onIconFrontClick', undefined);
+
     this.set('iconBack', 'close');
     assert
       .dom('span.tag')
       .hasClass('has-icon-back', 'DenaliTag has the `has-icon-back` class when iconBack is specified');
-    assert.dom('span.tag .d-icon').hasClass('d-close', 'test');
+    assert.dom('span.tag .d-icon').hasClass('d-close', 'DenaliTag has the specified icon in the back');
+
+    const iconBackClass = 'iconFrontClass';
```

Comment:
Would be nice if they matched
```suggestion
    const iconBackClass = 'iconBackClass';
```

Reply:
Good catch 👍